build/release/changelog.js

```diff
@@ -0,0 +1,163 @@
+import { writeFile } from "node:fs/promises";
+import { argv } from "node:process";
+import { exec as nodeExec } from "node:child_process";
+import util from "node:util";
+import { marked } from "marked";
+
+const exec = util.promisify( nodeExec );
+
+const rbeforeHash = /.#$/;
+const rendsWithHash = /#$/;
+const rcherry = / \(cherry picked from commit [^)]+\)/;
+const rcommit = /Fix(?:e[sd])? ((?:[a-zA-Z0-9_-]{1,39}\/[a-zA-Z0-9_-]{1,100}#)|#|gh-)(\d+)/g;
+const rcomponent = /^([^ :]+):\s*([^\n]+)/;
+const rnewline = /\r?\n/;
+
+const prevVersion = argv[ 2 ];
+const nextVersion = argv[ 3 ];
+
+if ( !prevVersion || !nextVersion ) {
+	throw new Error( "Usage: `node changelog.js PREV_VERSION NEXT_VERSION`" );
+}
+
+function ticketUrl( ticketId ) {
+	return `https://github.com/jquery/jquery/issues/${ ticketId }`;
+}
+
+function getTicketsForCommit( commit ) {
+	var tickets = [];
+
+	commit.replace(
```

Comment:
Is it weird? I thought it was convenient.

Reply:
It's like using `Array#map` without using its output when you really want `Array#forEach`. It obscures intent - `String#replace` is usually stateless, just like `Array#map`.

Why not use `String#matchAll` followed by `Array#map` with your logic?

Reply:
It feels weirder to me to do that because it's looping twice instead of once. Granted, a micro-optimization that doesn't matter for our sample size, but it's not the way I've trained my brain. 😛 I guess this is one of those cases where the ideal doesn't really exist. And yet, if there were some method that was more like `forEach`, the implementation of it might be about identical to `replace`.